### ISSUE TITLE
Default values are not expanded for `:str:` role

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#3787]
This is the output I see for `WPS211`:

```
» wps explain WPS211
WPS211 — Forbid too many arguments for a function or method.

Reasoning:
    This is an indicator of a bad design. When a function requires many
    arguments it is a sign that it should be refactored.
    It also indicates that the function does too many things at once.

Solution:
    Split the function into several functions.
    Then it will be easier to use them.

Configuration:
    This rule is configurable with ``--max-arguments``.
    Default: :str:`wemake_python_styleguide.options.defaults.MAX_ARGUMENTS`

.. versionadded:: 0.1.0
.. versionchanged:: 1.0.0
   Does not count special ``self``, ``cls``, and ``mcs`` as parameters.

   Also does not count parameters in ``@overload`` definitions.

See at website: https://pyflak.es/WPS211
                                        
```

Note that 

```
:str:`wemake_python_styleguide.options.defaults.MAX_ARGUMENTS`
```
is not expanded to be `5` https://github.com/wemake-services/wemake-python-styleguide/blob/664e75ce51cb2988cf69864e631f61cacb3a2794/wemake_python_styleguide/options/defaults.py#L75

But, it must be, right now both humans and LLMs will struggle with this.

**Comment by Khabib73** [comment]
What is the difference between [constants](https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/constants.py) and [default values](https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/options/defaults.py)? It seems both are used. Actually, I’ve almost finished, but I noticed that some default values are stored in [constants](https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/constants.py).

**Comment by sobolevn** [comment]
No difference, they should be rendered as well :)

**Comment by Khabib73** [comment]
One possible solution:

1. Collect all violations in a map, where the key is a string like `MAX_ARGUMENTS` and the value is the corresponding constant value. 
Like this:
```python
MAX_ARGUMENTS: Final = 5
...

DEFAULT_MAP = {
   'MAX_ARGUMENTS': MAX_ARGUMENTS
   ...
}
```
2. Find a pattern like this:
```text
Default: :str:`wemake_python_styleguide.options.defaults.MAX_ARGUMENTS`
```
3. Extract the default value key (e.g., `MAX_ARGUMENTS`) from this pattern.
4. Retrieve the real value from the map using that key.
5. Replace the original placeholder string with:
```text
Default: `REAL_VALUE`
```

Alternative solutions are also welcome—feel free to suggest ways to do this more simply and effectively :)

**Comment by sobolevn** [comment]
I would say that we can just prepare the replacement map and do `re.sub` :)

**Comment by Khabib73** [comment]
In wps explain, references like the one below are still printed as raw RST markup:
```
See
:py:data:`~wemake_python_styleguide.constants.FUNCTIONS_BLACKLIST`
for the full list of blacklisted functions.
```
Should we resolve these too and print the concrete values, or leave them as is? The lists are quite large (e.g. [VARIABLE_NAMES_BLACKLIST‎](https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/constants.py#L98)), so I'm unsure about the best format: one item per line, or a compact comma-separated single line?

**Comment by sobolevn** [comment]
Yes, good idea!
